Jimmy

yitaoli97 opened the pull request:
please review this screen

yitaoli97 committed:
added MessageManager attribute to UserController, updated registerForEvent/dropOutFromEvent parameters to match EventController, updated method parameters for sendMessage, added MessageManager to ProgramController

yitaoli97 committed:
implemented AttendeeScreen and AttendeeScreenController

bravetheheat commented on src/main/controllers/AttendeeScreenController.java:
```diff
@@ -0,0 +1,44 @@
+     * Checks valid input from attendee and tells ProgramController what screen to go next.
+     */
+    public String run() {
+        AttendeeScreen attendeeScreen = new AttendeeScreen();
```
You would likely want to make this a class variable

bravetheheat commented on src/main/controllers/AttendeeScreenController.java:
```diff
@@ -0,0 +1,44 @@
+     */
+    public String run() {
+        AttendeeScreen attendeeScreen = new AttendeeScreen();
+        String[] options = {"all events", "registered events", "contacts", "messages"};
```
I would do a numeric selection

bravetheheat commented on src/main/controllers/AttendeeScreenController.java:
```diff
@@ -0,0 +1,44 @@
+        String[] options = {"all events", "registered events", "contacts", "messages"};
+        List<String> prompts = (Arrays.asList(options));
+        attendeeScreen.prompt();
+        Scanner sc = new Scanner(System.in);
```
Make this a class variable

bravetheheat reviewed (commented):
The idea is there, so good work :)

yitaoli97 committed:
added tests for User classes and updated AttendeeScreen and AttendeeScreenController

yitaoli97 commented:
implemented suggested changes

yitaoli97 committed:
changes to how ProgramController switches screens

yitaoli97 commented:
changed the logic of switching screens in ProgramController
- added abstract ScreenController class
- starts from LoginScreenController
- run currentScreenController, then change currentScreeenController to the following ScreenController based on the output of currentScreenController
